src/expr/src/scalar/func.rs

```diff
@@ -2850,6 +2850,35 @@ fn substr<'a>(datums: &[Datum<'a>]) -> Datum<'a> {
     }
 }
 
+fn split_part<'a>(datums: &[Datum<'a>]) -> Result<Datum<'a>, EvalError> {
+    let string = datums[0].unwrap_str();
+    let delimiter = datums[1].unwrap_str();
+
+    // If the provided delimiter is the empty string,
+    // PostgreSQL does not break the string into individual
+    // characters. Instead, it generates the following parts: [string].
+    let parts = if delimiter == "" {
+        vec![string]
+    } else {
+        string.split(delimiter).collect()
+    };
+
+    // Provided index value begins at 1, not 0.
+    let index = datums[2].unwrap_int64() - 1;
+    if index < 0 {
+        return Err(EvalError::IntegerOutOfRange);
```

Comment:
This isn't quite the right error message. PostgreSQL returns "field position must be greater than zero."

Can you introduce a new `EvalError::InvalidParameterValue(String)` variant, and then instantiate with that error message?